homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -117,6 +117,9 @@ async def async_step_user(self, user_input=None):
                 user_input=user_input, errors=errors
             )
 
+        await self.async_set_unique_id(user_input[CONF_URL])
```

Comment:
This sets to different types of unique IDs to the config entry, which should generally be avoided. A manual entry does not require a unique ID in the flow.

Can't we get a more standardized/universal unique ID from the device itself? Like a serial number of some sort?

Reply:
There is no universally usable unique id available.
- serial exists, but is available only when the router is being used with username/pass, not in unauthenticated mode
- upnp udn exists, but only if upnp is enabled
- mac address is what we currently use for entity unique ids, but it has subnet related drawbacks